src/db/tables/onerep_scans.ts

```diff
@@ -62,6 +62,18 @@ async function setOnerepManualScan(
   });
 }
 
+async function updateOnerepScanStatus(
+  onerepScanId: number,
+  onerepScanStatus: Scan["status"]
+) {
+  await knex("onerep_scans").where("onerep_scan_id", onerepScanId).update({
+    onerep_scan_status: onerepScanStatus,
+    // @ts-ignore knex.fn.now() results in it being set to a date,
+    // even if it's not typed as a JS date object:
+    updated_at: knex.fn.now(),
+  });
```

Comment:
I wonder if it might make more sense to instead incorporate this into `addOnerepScanResults`? Since we basically only call that when we get an updated scan with a potentially updated status?

Reply:
I was wondering about that as well but was hesitant to change too much in that logic at first. I agree that this might be the more intuitive place to update the scan status. I’ve added [cb8a4aa](https://github.com/mozilla/blurts-server/pull/3470/commits/cb8a4aa442946be398e3fa85a5498024e13627f1) for either inserting or updating the scan.